nodejs/ece.js

```diff
@@ -240,19 +255,28 @@ function encryptRecord(key, counter, buffer, pad, padSize) {
   keylog('encrypt', buffer);
   pad = pad || 0;
   var nonce = generateNonce(key.nonce, counter);
-  var gcm = crypto.createCipheriv(AES_GCM, key.key, nonce);
   padSize = padSize || PAD_SIZE;
   var padding = new Buffer(pad + padSize);
   padding.fill(0);
   padding.writeUIntBE(pad, 0, padSize);
-  var epadding = gcm.update(padding);
-  var ebuffer = gcm.update(buffer);
-  gcm.final();
-  var tag = gcm.getAuthTag();
+
+  var plainText = Buffer.concat([padding, buffer]);
+  var result, tag;
+  if(!isOldNode) {
```

Comment:
Rather than split this, can I request that you make a shim for GCM that has a consistent interface.  Then you can implement it differently depending on node version.